src/main/java/org/gaul/s3proxy/Main.java

```diff
@@ -310,11 +310,13 @@ private static BlobStore createBlobStore(Properties properties,
         }
 
         BlobStoreContext context = builder.build(BlobStoreContext.class);
-        BlobStore blobStore = context.getBlobStore();
+        BlobStore blobStore;
```

Comment:
I don't oppose this change but how does it address your symptoms?  Wouldn't it have the same effect either way?

Reply:
So, I'd say that OVH swift misbehave when getBlobStore is called without any argument. But really I do not know internals of swift, nor jclouds.openstack.swift.
Here is a stacktrace from current latest docker image:

`[s3proxy] W 03-06 10:26:48.936 main o.j.l.s.i.GetRegionIdMatchingProviderURIOrNull:74 |::] failed to find key for value https://auth.cloud.ovh.net/v3 in {DE1=https://storage.de1.cloud.ovh.net/v1/AUTH_d446...43, UK1=https://storage.uk1.cloud.ovh.net/v1/AUTH_d446...43, WAW1=https://storage.waw1.cloud.ovh.net/v1/AUTH_d446...43, GRA5=https://storage.gra5.cloud.ovh.net/v1/AUTH_d446...43, BHS3=https://storage.bhs3.cloud.ovh.net/v1/AUTH_d446...43, SBG5=https://storage.sbg5.cloud.ovh.net/v1/AUTH_d446...43}; choosing first: DE1
[s3proxy] E 03-06 10:26:48.941 main org.gaul.s3proxy.Main:236 |::] Exception in thread "main" 
[s3proxy] E 03-06 10:26:48.941 main org.gaul.s3proxy.Main:236 |::] java.lang.IllegalArgumentException: region DE1 not in [UK1]
[s3proxy] E 03-06 10:26:48.942 main org.gaul.s3proxy.Main:236 |::] 	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:145)
[s3proxy] E 03-06 10:26:48.942 main org.gaul.s3proxy.Main:236 |::] 	at org.jclouds.openstack.swift.v1.blobstore.RegionScopedBlobStoreContext.checkRegionId(RegionScopedBlobStoreContext.java:88)
[s3proxy] E 03-06 10:26:48.942 main org.gaul.s3proxy.Main:236 |::] 	at org.jclouds.openstack.swift.v1.blobstore.RegionScopedBlobStoreContext.getBlobStore(RegionScopedBlobStoreContext.java:72)
[s3proxy] E 03-06 10:26:48.942 main org.gaul.s3proxy.Main:236 |::] 	at org.jclouds.openstack.swift.v1.blobstore.RegionScopedBlobStoreContext.getBlobStore(RegionScopedBlobStoreContext.java:122)
[s3proxy] E 03-06 10:26:48.942 main org.gaul.s3proxy.Main:236 |::] 	at org.gaul.s3proxy.Main.createBlobStore(Main.java:313)
[s3proxy] E 03-06 10:26:48.943 main org.gaul.s3proxy.Main:236 |::] 	at org.gaul.s3proxy.Main.main(Main.java:112)`

>Wouldn't it have the same effect either way?
After my change there is no error, region is chosen properly, so it works differently.

Reply:
Now I understand -- calling `context.getBlobStore` has a side-effect.